generic_config_updater/gu_common.py

```diff
@@ -63,13 +64,16 @@ def get_config_db_as_json(self):
         return config_db_json
 
     def _get_config_db_as_text(self):
-        # TODO: Getting configs from CLI is very slow, need to get it from sonic-cffgen directly
-        cmd = "show runningconfiguration all"
-        result = subprocess.Popen(cmd, shell=True, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        if self.namespace is not None and self.namespace != multi_asic.DEFAULT_NAMESPACE:
+            cmd = ['sonic-cfggen', '-d', '--print-data', '-n', self.namespace]
+        else:
+            cmd = ['sonic-cfggen', '-d', '--print-data']
+
+        result = subprocess.Popen(cmd, shell=False, text=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         text, err = result.communicate()
         return_code = result.returncode
         if return_code: # non-zero means failure
-            raise GenericConfigUpdaterError(f"Failed to get running config, Return code: {return_code}, Error: {err}")
+            raise GenericConfigUpdaterError(f"Failed to get running config for namespace: {self.namespace}, Return code: {return_code}, Error: {err}")
         return text
```

Comment:
change the assertion error message to have localhost when namespace is `DEFAULT_NAMSPACE`?

Reply:
No, if the namespace is `DEFAULT_NAMSPACE`, for here, it will be empty instead of `localhost`